### PR TITLE
Add deprecation warning for s3_data_distribution_type in Clarify Data Config

### PR DESCRIPTION
**Pull request by keerthanvasist**
*Description of changes:*
In Clarify DataConfig, there is a parameter called `s3_data_distribution_type` which can currently take two values: `FullyReplicated` or `ShardedByS3Key`. However, going forward, the only valid option for this variable will be `FullyReplicated`. This CR also adds a deprecation warning to the class `DataConfig` so that customers get time to make changes to their code, as we will eventually remove this parameter (which will be a backward incompatible change). 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/test_clarify.py [hunk 1]
```diff
@@ -548,10 +548,18 @@ def test_pre_training_bias(
     name_from_base,
     clarify_processor,
     clarify_processor_with_job_name_prefix,
-    data_config,
     data_bias_config,
 ):
     with patch.object(SageMakerClarifyProcessor, "_run", return_value=None) as mock_method:
+        data_config = DataConfig(
```
Comment (larroy) [inline tests/unit/test_clarify.py:554]: Please add a comment that the value is ignored, shall we capture logging and verify we throw a warning? What concerns me about this test is that I don't see any check that there's no side effect, even if it's a minor thing. Maybe clarify on a comment what's the expectation of the test? is not clear.

Comment (shreyapandit) [reply tests/unit/test_clarify.py:554]: +1

Comment (keerthanvasist) [reply tests/unit/test_clarify.py]: I updated have added a test that explicitly verifies that any value other than "FullyReplicated" is ignored.

**Review by larroy — APPROVED** [review]

**Review by jeniyat — APPROVED** [review]

**Review by shreyapandit — APPROVED** [review]

**Comment by melanie531** [comment]
> _Description of changes:_ In Clarify DataConfig, there is a parameter called `s3_data_distribution_type` which can currently take two values: `FullyReplicated` or `ShardedByS3Key`. However, going forward, the only valid option for this variable will be `FullyReplicated`. This CR also adds a deprecation warning to the class `DataConfig` so that customers get time to make changes to their code, as we will eventually remove this parameter (which will be a backward incompatible change).
> 
> _Testing done:_
> 
> ## Merge Checklist
> _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._
> 
> #### General
> * [x]  I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
> * [x]  I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
> * [x]  I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
> * [x]  I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
> * [x]  I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)
> 
> #### Tests
> * [x]  I have added tests that prove my fix is effective or that my feature works (if appropriate)
> * [x]  I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
> * [x]  I have checked that my tests are not configured for a specific region or account (if appropriate)
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Hi team, this deprecated warning message is a bit confuse. It reads like "DataConfig will be deprecated on 15 Mar 2022.s3_data_distribution_type parameter will no longer be supported." But the DataConfig is not deprecated right?

**Comment by keerthanvasist** [comment]
Yes. `DataConfig` is not deprecated. I have created a PR to remove it here https://github.com/aws/sagemaker-python-sdk/pull/3064. I need help with this PR. The tests are failing but aren't related. Can you please help?

**Comment by melanie531** [comment]
> 

Hi keerthanvasist@, I'm not part of the SageMaker SDK development team. Would love to help if I can. Please feel free to let me know if anything I can do to help here.

**Pull request merged**